src/concurrent_server.py

```diff
@@ -0,0 +1,104 @@
+# _*_ coding: utf-8 _*_
+"""Create a server that can handle concurrent messages."""
+import os
+import io
+import mimetypes
+
+
+def server(socket, address):
+
+    msg = b''
+    buffer_length = 8
+    message_complete = False
+    while not message_complete:
+        data = socket.recv(buffer_length)
+        msg += data
+        if len(data) < buffer_length:
+            break
+    msg = msg.decode('utf-8')
+    try:
+        path = parse_request(msg)
+        get_stuff = resolve_uri(path)
+        response = response_ok(get_stuff)
+    except NameError:
+        response = response_error("405", "Method not allowed")
+    except AttributeError:
+        response = response_error("403", "Forbidden")
+    except LookupError:
+        response = response_error("400", "Bad Request")
+    except IOError:
+        response = response_error("404", "File Not Found")
+    socket.sendall(response)
+    socket.close()
+
+
+def parse_request(argument):
+    """Check whether message is proper HTTP request."""
+    request_bits = argument.split('\n')
```

Comment:
You really want to use a "\r\n" to avoid any possibility of splitting in the wrong places.